CLOUDSTACK-10333: Secure Live VM Migration for KVM

rhtyd opened the pull request:
## Description

This extends securing of KVM hosts to securing of libvirt on KVM
host as well for TLS enabled live VM migration. To simplify implementation
securing of host implies that both host and libvirtd processes are
secured with management server's CA plugin issued certificates.

FS: https://cwiki.apache.org/confluence/display/CLOUDSTACK/Secure+Live+VM+Migration+for+KVM

Based on whether keystore and certificates files are available at
/etc/cloudstack/agent, the KVM agent determines whether to use TLS or
TCP based uris for live VM migration. It is also enforced that a secured
host will allow live VM migration to/from other secured host, and an
unsecured hosts will allow live VM migration to/from other unsecured
host only.

Post upgrade the KVM agent on startup will expose its security state
(secured detail is sent as true or false) to the managements server that
gets saved in host_details for the host. This host detail can be accesed
via the listHosts response, and in the UI unsecured KVM hosts will show
up with the host state of ‘unsecured’. Further, a button has been added
that allows admins to provision/renew certificates to KVM hosts and can
be used to secure any unsecured KVM host.

The `cloudstack-setup-agent` was modified to accept a new flag `-s`
which will reconfigure libvirtd with following settings:

    listen_tcp=0
    listen_tls=1
    tcp_port="16509"
    tls_port="16514"
    auth_tcp="none"
    auth_tls="none"
    key_file = "/etc/pki/libvirt/private/serverkey.pem"
    cert_file = "/etc/pki/libvirt/servercert.pem"
    ca_file = "/etc/pki/CA/cacert.pem"

For a connected KVM host agent, when the certificate are
renewed/provisioned a background task is scheduled that waits until all
of the agent tasks finish after which libvirt process is restarted and
finally the agent is restarted via AgentShell.

There are no API or DB changes.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Screenshots (if appropriate):

Unsecured KVM hosts when are Up and connected show up as:
![screenshot from 2018-03-20 19-18-42](https://user-images.githubusercontent.com/95203/37759459-b0f31564-2dd9-11e8-9fc2-9a16259c750e.png)

Button to secure host, or renew/provision certificate using configured CA plugin;
![screenshot from 2018-03-20 19-18-49](https://user-images.githubusercontent.com/95203/37759461-b1383ba8-2dd9-11e8-8770-d8a07a52e124.png)
![screenshot from 2018-03-20 19-19-00](https://user-images.githubusercontent.com/95203/37759462-b184777a-2dd9-11e8-958e-d06197a5bf69.png)

Error message when VM migration is between secured and unsecured hosts:
![screenshot from 2018-03-20 19-20-48](https://user-images.githubusercontent.com/95203/37759463-b1ca3742-2dd9-11e8-8ec7-1f866ea5d87a.png)

## How Has This Been Tested?

- Tested between a pre-upgraded secured KVM host (el7 based) and post-upgrade it secures libvirtd using same certificates.
- Tested tls enabled live vm migration between two secured and unsecured el7 kvm hosts.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly. PR https://github.com/apache/cloudstack-docs-admin/pull/50, https://github.com/apache/cloudstack-docs-install/pull/36
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

<!-- The following will kick a packaging job, remove if as applicable -->
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1807

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1808

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-2401)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 24349 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2505-t2401-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_deploy_virtio_scsi_vm.py
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermitten failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 65 look OK, 2 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_rvpc_network_garbage_collector_nics | `Failure` | 448.47 | test_vpc_redundant.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 2.09 | test_hostha_kvm.py

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-1820

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-1821

wido commented on debian/cloudstack-agent.postinst:
```diff
@@ -50,6 +50,13 @@ case "$1" in
+            iptables -t filter -A INPUT -p tcp -m tcp --dport 16514 -j ACCEPT
+            iptables-save > /etc/iptables/rules.v4
+        fi
+
```
I am not to keen on this one. Do we really want our packages to start configuring a firewall on a host?

This should be on the docs to tell people to open the port(s), not having packages doing it manually.

In addition, the directory /etc/iptables does not exist by default on Ubuntu.

rhtyd replied:
The `cloudstack-setup-agent` reconfigures libvirtd config and also adds iptables rules for several ports, the post-install script (both rpm+deb) does a test if expected iptables rules are in place and adds the ACCEPT rule only if needed.

Given not all users may use a config mgmt system such as chef/puppet/ansible, running the commands as part of post-install script will save the additional work they may need to do themselves (manually or automated). @wido I'm okay to advise users via release/admin docs, but I don't see any negative/side-effects with the change.

wido replied:
If users are using firewalld or ufw on their CentOS/Ubuntu system this may break things. And like I said, /etc/iptables does not exist on Ubuntu systems by default, you need the iptables-persistent package for that.

I wouldn't touch the firewall in a postinst of a package. The package should not touch parts of the system it's not configuring.

rhtyd replied:
Alright @wido, perhaps we can remove this for Ubuntu (debian pkg). On both CentOS 6 and 7, iptables service is indeed available that is used to save existing rules, firewalld is not used here. It is likely that things may break for el6/7 users. I'm okay to document the change in release notes docs as well. Let's ask others for their thoughts - @DaanHoogland @rafaelweingartner @resmo @ustcweizhou @nvazquez @mlsorensen ?

DaanHoogland replied:
Is there an alternative that will work. I would not like to see a large devide between the amount of work to be done on rhel-like systems versus debian-likes.

rhtyd replied:
Okay, removed this.

wido commented on packaging/centos63/cloud.spec:
```diff
@@ -493,6 +493,12 @@ if [ -f "%{_sysconfdir}/cloud.rpmsave/agent/agent.properties" ]; then
+    iptables -t filter -A INPUT -p tcp -m tcp --dport 16514 -j ACCEPT
+    iptables-save > /etc/iptables/rules.v4
+fi
+
```
Same here as in the DEB package, I'm not in favor of this

rhtyd replied:
We'll document in release notes and admin/install docs. Fixed.

wido commented on packaging/centos7/cloud.spec:
```diff
@@ -437,6 +437,12 @@ if [ -f "%{_sysconfdir}/cloud.rpmsave/agent/agent.properties" ]; then
+if ! iptables-save | grep -- "-A INPUT -p tcp -m tcp --dport 16514 -j ACCEPT" > /dev/null; then
+    iptables -t filter -A INPUT -p tcp -m tcp --dport 16514 -j ACCEPT
+    iptables-save > /etc/iptables/rules.v4
+fi
```
And here again

rhtyd replied:
We'll document in release notes and admin/install docs. Fixed.

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-1830

wido commented:
Although the iptables changes have been removed for Ubuntu/Debian I think we should also remove them from the RPM packages.

In my opinion a package should never be allowed to touch firewalls without the operating knowing it.

In the documentation we already tell users to open ports: http://docs.cloudstack.apache.org/projects/cloudstack-installation/en/4.11/hypervisor/kvm.html#configuring-the-firewall

If additional ports need to be opened we should put them in there, but not just open them in a RPM or DEB package.

borisstoyanov commented:
@rhtyd there seems to be some build errors with the latests changes

rhtyd commented:
@borisstoyanov yes there was an unused import.

@wido bear in mind that `cloudstack-setup-agent` will reconfigure network, libvirtd, iptables configuration. It's a little known fact, and not well documented; the docs are redundant (they tell you to add stuff, that cloudstack-setup-agent already does, you may test it yourself). I've removed the post-install steps, now you can re-review. I've sent doc PRs instead:
apache/cloudstack-docs-admin#50
apache/cloudstack-docs-install#36

rhtyd commented:
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1836

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

wido commented:
@rhtyd Yes, that is correct. That tool is allowed to make such changes, but we should not just do this in a postinst of a package.

The postinst of a package is to clean up old files related to the package, handle some things around the package, but it should not start touching firewalling on a host.

That is just not done when it comes to packages.

borisstoyanov commented:
@rhtyd I'm trying to migrate a VM between unsecured hosts and got the following exception: 
```
2018-03-28 12:31:52,414 ERROR [c.c.v.VmWorkJobHandlerProxy] (Work-Job-Executor-6:ctx-bc927fcc job-38/job-39 ctx-441b0b18) (logid:d8b1d686) Invocation exception, caused by: com.cloud.utils.exception.CloudRuntimeException: org.libvirt.LibvirtException: Cannot get interface MTU on 'breth0-769': No such device
2018-03-28 12:31:52,415 INFO  [c.c.v.VmWorkJobHandlerProxy] (Work-Job-Executor-6:ctx-bc927fcc job-38/job-39 ctx-441b0b18) (logid:d8b1d686) Rethrow exception com.cloud.utils.exception.CloudRuntimeException: org.libvirt.LibvirtException: Cannot get interface MTU on 'breth0-769': No such device
2018-03-28 12:31:52,415 DEBUG [c.c.v.VmWorkJobDispatcher] (Work-Job-Executor-6:ctx-bc927fcc job-38/job-39) (logid:d8b1d686) Done with run of VM work job: com.cloud.vm.VmWorkMigrate for VM 3, job origin: 38
2018-03-28 12:31:52,415 ERROR [c.c.v.VmWorkJobDispatcher] (Work-Job-Executor-6:ctx-bc927fcc job-38/job-39) (logid:d8b1d686) Unable to complete AsyncJobVO {id:39, userId: 2, accountId: 2, instanceType: null, instanceId: null, cmd: com.cloud.vm.VmWorkMigrate, cmdInfo: rO0ABXNyABpjb20uY2xvdWQudm0uVm1Xb3JrTWlncmF0ZRdxQXtPtzYqAgAGSgAJc3JjSG9zdElkTAAJY2x1c3RlcklkdAAQTGphdmEvbGFuZy9Mb25nO0wABmhvc3RJZHEAfgABTAAFcG9kSWRxAH4AAUwAB3N0b3JhZ2V0AA9MamF2YS91dGlsL01hcDtMAAZ6b25lSWRxAH4AAXhyABNjb20uY2xvdWQudm0uVm1Xb3Jrn5m2VvAlZ2sCAARKAAlhY2NvdW50SWRKAAZ1c2VySWRKAAR2bUlkTAALaGFuZGxlck5hbWV0ABJMamF2YS9sYW5nL1N0cmluZzt4cAAAAAAAAAACAAAAAAAAAAIAAAAAAAAAA3QAGVZpcnR1YWxNYWNoaW5lTWFuYWdlckltcGwAAAAAAAAAAXNyAA5qYXZhLmxhbmcuTG9uZzuL5JDMjyPfAgABSgAFdmFsdWV4cgAQamF2YS5sYW5nLk51bWJlcoaslR0LlOCLAgAAeHAAAAAAAAAAAXNxAH4ABwAAAAAAAAACcQB-AAlwcQB-AAk, cmdVersion: 0, status: IN_PROGRESS, processStatus: 0, resultCode: 0, result: null, initMsid: 6653541943041, completeMsid: null, lastUpdated: null, lastPolled: null, created: Wed Mar 28 12:31:50 UTC 2018}, job origin:38
com.cloud.utils.exception.CloudRuntimeException: org.libvirt.LibvirtException: Cannot get interface MTU on 'breth0-769': No such device
```
I've made the hosts unsecured by deleting the executing the cloudstack-setup-agent script without the -s option.

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

borisstoyanov commented:
Maybe this ^^ needs further investigation since I've provisioned the certificates on both the hosts and still getting the same error.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1848

blueorangutan commented:
<b>Trillian test result (tid-2427)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 126331 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2505-t2427-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_deploy_virtio_scsi_vm.py
Intermitten failure detected: /marvin/tests/smoke/test_internal_lb.py
Intermitten failure detected: /marvin/tests/smoke/test_public_ip_range.py
Intermitten failure detected: /marvin/tests/smoke/test_routers.py
Intermitten failure detected: /marvin/tests/smoke/test_templates.py
Intermitten failure detected: /marvin/tests/smoke/test_usage.py
Intermitten failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermitten failure detected: /marvin/tests/smoke/test_volumes.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermitten failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 60 look OK, 7 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_restart_network_wo_cleanup | `Failure` | 4.25 | test_routers.py
test_04_extract_template | `Failure` | 128.35 | test_templates.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
test_01_secured_vm_migration | `Failure` | 934.74 | test_vm_life_cycle.py
test_02_not_secured_vm_migration | `Error` | 0.22 | test_vm_life_cycle.py
test_03_secured_to_nonsecured_vm_migration | `Error` | 1.26 | test_vm_life_cycle.py
test_04_nonsecured_to_secured_vm_migration | `Error` | 1.26 | test_vm_life_cycle.py
test_06_download_detached_volume | `Failure` | 143.42 | test_volumes.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 505.57 | test_vpc_redundant.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 2.24 | test_hostha_kvm.py

rhtyd commented:
@borisstoyanov as discovered in lab, the issue was related to traffic label and was env related. Please use the sed based unsecuring approach, and avoid cloudstack-setup-agent to speed up tests.
@blueorangutan package

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1849

rhtyd commented:
@blueorangutan test

blueorangutan commented:
@rhtyd a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

rafaelweingartner commented on agent/src/com/cloud/agent/Agent.java:
```diff
@@ -140,9 +143,11 @@ public int value() {
+    ThreadPoolExecutor _ugentTaskPool;
     ExecutorService _executor;
 
+    Thread _shutdownThread = new ShutdownThread(this);
```
Would you mind not introducing `_` in the code anymore?
They are meaningless in our code base.

rhtyd replied:
I'm not fan of them either, but added the name to keep the naming consistent in the Agent class. Sure, will fix this for all the variables in Agent then.

rafaelweingartner replied:
Awesome! Thanks!
This is a cosmetic thing, but if we do not get ride of them, newcomers might think that we enforce them. I am trying to do the same in other PRs.

rafaelweingartner commented on agent/src/com/cloud/agent/Agent.java:
```diff
@@ -274,6 +279,19 @@ public void start() {
         _shell.updateConnectedHost();
+
+        // In case of software based restart, GC to remove old instances
+        _executor.submit(new Runnable() {
```
Can you explaining the goal of this code here?
Remove old instance of Java objects?

rhtyd replied:
Yes @rafaelweingartner, with the feature we've introduced a background thread that will perform software based restart of the agent. Look at the post renewal restart task. With this it will be easier to restart an agent without actually restarting the agent JVM process. This runnable is needed to GC old agent instance, we can remove this as well, but keeping it ensures that old agent is stopped+GC-ed.

rafaelweingartner replied:
Got it. There is only one thing. We never now for sure that the GC has been executed. `System.gc` only suggests for the JVM that the GC can be executed, but there is no guarantees. Anyways, as long as the old objects are not used anymore we are safe to go.

Would you mind extracting this block of code to a method and document it with this explanation?

rhtyd replied:
Fixed.

borisstoyanov commented:
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

borisstoyanov commented:
I'll kick new tests with the latest commit once the build comes out

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-1850

borisstoyanov commented:
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✖centos6 ✖centos7 ✖debian. JID-1851

borisstoyanov commented:
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1855

borisstoyanov commented:
@blueorangutan test

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-2459)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 107420 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2505-t2459-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_routers_network_ops.py
Intermitten failure detected: /marvin/tests/smoke/test_routers.py
Intermitten failure detected: /marvin/tests/smoke/test_ssvm.py
Intermitten failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Smoke tests completed. 65 look OK, 2 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_restart_network_wo_cleanup | `Failure` | 4.14 | test_routers.py
test_01_secured_vm_migration | `Failure` | 1069.92 | test_vm_life_cycle.py
test_02_not_secured_vm_migration | `Failure` | 118.28 | test_vm_life_cycle.py

rhtyd commented:
Looks like there is an outstanding/related issue to CA management, I'll have a look next week (on holidays now).

DaanHoogland commented on test/integration/smoke/test_vm_life_cycle.py:
```diff
@@ -21,9 +21,11 @@
-                                  detachIso)
-from marvin.lib.utils import (cleanup_resources,
-                              validateList)
+                                  detachIso,
```
strange alignment is happening here and below. not fatal, just a headsup

borisstoyanov commented:
I've resolved the conflict, let me run tests again @rhtyd 
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

rhtyd commented:
Okay @borisstoyanov 
All - please hold merging this, I may include some keystore related changes reported in recent issues.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1902

borisstoyanov commented:
@blueorangutan test

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-2494)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 101305 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2505-t2494-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_routers.py
Intermitten failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermitten failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 63 look OK, 4 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_restart_network_wo_cleanup | `Failure` | 3.99 | test_routers.py
test_01_secured_vm_migration | `Failure` | 1033.48 | test_vm_life_cycle.py
test_02_not_secured_vm_migration | `Failure` | 193.32 | test_vm_life_cycle.py
test_04_rvpc_network_garbage_collector_nics | `Failure` | 287.33 | test_vpc_redundant.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 1.39 | test_hostha_kvm.py

DaanHoogland commented:
As you announced extra commits , i'm marking this WIP @rhtyd

borisstoyanov commented:
@blueorangutan package

blueorangutan commented:
@borisstoyanov a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1918

borisstoyanov commented:
@blueorangutan test

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-2502)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 112688 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2505-t2502-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_outofbandmanagement_nestedplugin.py
Intermitten failure detected: /marvin/tests/smoke/test_privategw_acl.py
Intermitten failure detected: /marvin/tests/smoke/test_routers.py
Intermitten failure detected: /marvin/tests/smoke/test_vm_life_cycle.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_redundant.py
Intermitten failure detected: /marvin/tests/smoke/test_vpc_vpn.py
Intermitten failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 64 look OK, 3 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_restart_network_wo_cleanup | `Failure` | 4.25 | test_routers.py
test_01_secured_vm_migration | `Error` | 23.83 | test_vm_life_cycle.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 2.63 | test_hostha_kvm.py

borisstoyanov commented:
@blueorangutan test

blueorangutan commented:
@borisstoyanov a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-2514)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 107105 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr2505-t2514-kvm-centos7.zip
Intermitten failure detected: /marvin/tests/smoke/test_public_ip_range.py
Intermitten failure detected: /marvin/tests/smoke/test_routers.py
Intermitten failure detected: /marvin/tests/smoke/test_templates.py
Intermitten failure detected: /marvin/tests/smoke/test_usage.py
Intermitten failure detected: /marvin/tests/smoke/test_volumes.py
Intermitten failure detected: /marvin/tests/smoke/test_hostha_kvm.py
Smoke tests completed. 62 look OK, 5 have error(s)
Only failed tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_04_restart_network_wo_cleanup | `Failure` | 2.96 | test_routers.py
test_04_extract_template | `Failure` | 128.34 | test_templates.py
ContextSuite context=TestISOUsage>:setup | `Error` | 0.00 | test_usage.py
test_06_download_detached_volume | `Failure` | 137.61 | test_volumes.py
test_hostha_enable_ha_when_host_in_maintenance | `Error` | 2.48 | test_hostha_kvm.py

rhtyd commented:
Addressed code review comment, will merge this as soon as travis is green. Tests pass:
```
Test secured VM migration ... === TestName: test_01_secured_vm_migration | Status : SUCCESS ===
ok
Test Non-secured VM Migration ... === TestName: test_02_not_secured_vm_migration | Status : SUCCESS ===
ok
Test destroy Virtual Machine ... === TestName: test_03_secured_to_nonsecured_vm_migration | Status : SUCCESS ===
ok
Test Non-secured VM Migration ... === TestName: test_04_nonsecured_to_secured_vm_migration | Status : SUCCESS ===
```

rhtyd commented:
All looks good. I'll merge this based on code reviews and tests. Doc PRs have been merged as well.

blueorangutan commented:
@rhtyd a Jenkins job has been kicked to build packages. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: ✔centos6 ✔centos7 ✔debian. JID-1962
